File: python/openclaw-maintenance.py
#!/usr/bin/env python3
# openclaw-maintenance.sh — portiert nach python
# Quelle: shell, OpenClaw@gateway1:scripts/openclaw-maintenance.sh
# Erzeugt: 2026-08-09 durch ABSTRACTIONS_MANAGER.py

import subprocess
import sys

def run_command(command):
    """Führt einen Shell-Befehl aus und gibt den Rückgabewert zurück."""
    try:
        result = subprocess.run(command, shell=True, check=True, capture_output=True, text=True)
        return result.returncode, result.stdout, result.stderr
    except subprocess.CalledProcessError as e:
        return e.returncode, e.stdout, e.stderr

# === 1. Service-/Config-Drift ===
run_command("openclaw doctor")

# === 2. Plugin-Stage (aktive Varianten, NICHT plugins doctor) ===
run_command("openclaw plugins registry --refresh")
run_command("openclaw plugins update --all")

# === 3. Tasks ===
run_command("openclaw tasks maintenance --apply")

# === 4. Sessions – alle Agents auf einmal ===
run_command("openclaw sessions cleanup --enforce --all-agents")

# === 5. Memory – status/index decken alle Agents ab ===
run_command("openclaw memory status --deep --fix")
run_command("openclaw memory index --force")

# === 6. Memory promote – MUSS pro Agent ===
agents = ["main", "knecht", "docs", "ops-hub", "cron"]
for agent in agents:
    run_command(f'openclaw memory promote --apply --agent "{agent}"')

# === 7. Secrets ===
run_command("openclaw secrets reload")

File: javascript/openclaw-maintenance.js
#!/usr/bin/env node
// openclaw-maintenance.sh — portiert nach javascript
// Quelle: shell, OpenClaw@gateway1:scripts/openclaw-maintenance.sh
// Erzeugt: 2026-08-09 durch ABSTRACTIONS_MANAGER.py

import { execSync } from 'child_process';
import { stdout } from 'process';

// === 1. Service-/Config-Drift ===
execSync('openclaw doctor', { stdio: 'inherit' });

// === 2. Plugin-Stage (aktive Varianten, NICHT plugins doctor) ===
execSync('openclaw plugins registry --refresh', { stdio: 'inherit' });
execSync('openclaw plugins update --all', { stdio: 'inherit' });

// === 3. Tasks ===
execSync('openclaw tasks maintenance --apply', { stdio: 'inherit' });

// === 4. Sessions – alle Agents auf einmal ===
execSync('openclaw sessions cleanup --enforce --all-agents', { stdio: 'inherit' });

// === 5. Memory – status/index decken alle Agents ab ===
execSync('openclaw memory status --deep --fix', { stdio: 'inherit' });
execSync('openclaw memory index --force', { stdio: 'inherit' });

// === 6. Memory promote – MUSS pro Agent ===
const agents = ['main', 'knecht', 'docs', 'ops-hub', 'cron'];
for (const agent of agents) {
  execSync(`openclaw memory promote --apply --agent "${agent}"`, { stdio: 'inherit' });
}

// === 7. Secrets ===
execSync('openclaw secrets reload', { stdio: 'inherit' });
